sdk/core/core-http/src/util/xml.browser.ts

```diff
@@ -8,17 +8,39 @@ if (!self.document || !self.DOMParser || !self.Node || !self.XMLSerializer) {
     `This library depends on the following DOM objects: ["document", "DOMParser", "Node", "XMLSerializer"] to parse XML, but some of these are undefined. You may provide a polyfill to make these globally available in order to support your environment. For more information, please refer to https://aka.ms/azsdk/js/web-workers. `
   );
 }
-const doc = document.implementation.createDocument(null, null, null);
 
-const parser = new DOMParser();
+let cachedDoc: Document | undefined;
+function getDoc(): Document {
```

Comment:
Hey I just saw this change. Won't the check on line 6 still throw an early error? https://github.com/Azure/azure-sdk-for-js/pull/21118/files#diff-3cbda4a55114f22779fe6c5836f20cb6f66d3f6332571692b8b49014aa52d6d3R6

For context, at the time (so long ago now 🥲 ) we had discussed an early vs later error https://github.com/Azure/azure-sdk-for-js/pull/13977#issuecomment-786287682

But maybe react-native changes those requirements?

Reply:
It still throws! But I can work around it by setting some empty objects to global for now when I am sure the library doesn't use xml.  I will work on adding proper xml support later.